### PR TITLE
The `\name{}` field in an Rd file shouldn't be used in the topic index

### LINKED ISSUE #1576: autolinking and rdname
https://github.com/r-lib/usethis/issues/1392

### PR DESCRIPTION
**Pull request by dmurdoch**
Fix #1576 

The `topic_index` is documented here:  https://github.com/r-lib/downlit/blob/e22f072cfeaf91fbe7c38aeabd351aaa184d36fd/R/topic-index.R#L2, to map "aliases to Rd file names (sans extension)".  The `\name{}` field can differ, so it shouldn't be used here. 

This fixes https://github.com/r-lib/downlit/issues/83 .

### CODE REVIEW

**Comment by hadley** [comment]
@maelle does this fix https://github.com/r-lib/pkgdown/issues/1576 ?

**Comment by maelle** [comment]
Yes, it does, I've just tested it, and it's similar to [`3b5a205` (#1577)](https://github.com/r-lib/pkgdown/pull/1577/commits/3b5a205d7c9e72680f53062547a81cb6ae089abd) that worked.

File: R/context.R [hunk 1]
```diff
@@ -10,7 +10,8 @@ section_init <- function(pkg, depth, override = list(), .frame = parent.frame())
 
 local_options_link <- function(pkg, depth, .frame = parent.frame()) {
   article_index <- set_names(path_file(pkg$vignettes$file_out), pkg$vignettes$name)
-  topic_index <- invert_index(set_names(pkg$topics$alias, pkg$topics$name))
+  Rdname <- gsub(".Rd$", "", pkg$topic$file_in)
```
Comment (maelle) [inline R/context.R:13]:
```suggestion
  Rdname <- fs::path_ext_remove(pkg$topic$file_in)
```

Comment (maelle) [reply R/context.R:13]: Since pkdown depends on fs anyway. :slightly_smiling_face:

**Comment by hadley** [comment]
Looks like I got a bit confused between issues and repos here 😬

**Review by hadley — APPROVED** [review]
Looks good, thanks! Can you please add a bullet to the top of `NEWS.md`? It should briefly [describe the change](http://style.tidyverse.org/news.html) and end with `(@yourname, #issuenumber)`.

**Comment by hadley** [comment]
Thanks @dmurdoch!

**Pull request merged**